examples/structured_2d_dgsem/elixir_euler_rayleigh_taylor_instability.jl

```diff
@@ -0,0 +1,68 @@
+
+using OrdinaryDiffEq
+using Trixi
+
+###############################################################################
+# semidiscretization of the compressible Euler equations
+gamma = 1.4
+equations = CompressibleEulerEquations2D(gamma)
+
+volume_flux  = flux_ranocha
+solver = DGSEM(polydeg=3, surface_flux=flux_hll,
+               volume_integral=VolumeIntegralFluxDifferencing(volume_flux))
+
+# The domain is [0, .25] x [0, 1]
+mapping(xi, eta) = SVector(.25 * .5 * (1 + xi), .5 * (1 + eta))
+
+num_elements_per_dimension = 32
+cells_per_dimension = (num_elements_per_dimension, num_elements_per_dimension * 4)
+mesh = StructuredMesh(cells_per_dimension, mapping)
+
+boundary_condition_wall = BoundaryConditionWall(boundary_state_slip_wall)
+boundary_conditions = (
+                       x_neg=boundary_condition_wall,
+                       x_pos=boundary_condition_wall,
+                       y_neg=boundary_condition_wall,
+                       y_pos=boundary_condition_wall,
+                      )
+
+## Alternative setup: left/right periodic BCs and Dirichlet BCs on the top/bottom.
+# boundary_conditions = (
+#                        x_neg=boundary_condition_periodic,
+#                        x_pos=boundary_condition_periodic,
+#                        y_neg=BoundaryConditionDirichlet(initial_condition),
+#                        y_pos=BoundaryConditionDirichlet(initial_condition),
+#                       )
+
+initial_condition = initial_condition_rayleigh_taylor_instability
+semi = SemidiscretizationHyperbolic(mesh, equations, initial_condition, solver;
+                                    source_terms=source_terms_rayleigh_taylor_instability,
+                                    boundary_conditions=boundary_conditions)
+
+###############################################################################
+# ODE solvers, callbacks etc.
+
+tspan = (0.0, 1.5)
+ode = semidiscretize(semi, tspan)
+
+summary_callback = SummaryCallback()
+
+analysis_interval = 100
+analysis_callback = AnalysisCallback(semi, interval=analysis_interval)
+
+alive_callback = AliveCallback(analysis_interval=analysis_interval)
+
+stepsize_callback = StepsizeCallback(cfl=1.3)
+
+callbacks = CallbackSet(summary_callback,
+                        analysis_callback,
+                        alive_callback,
+                        stepsize_callback)
+
+###############################################################################
+# run the simulation
+
+sol = solve(ode, RDPK3SpFSAL49(), abstol=1.0e-6, reltol=1.0e-6,
+            save_everystep=false, callback=callbacks);
```

Comment:
I would need to double check (which I can't do right now on my phone) but I think the step size callback will basically do nothing when applied to an adaptive time step method such as `RDPK3SpFSAL49()`. If that's true, you should either remove the `stepsize_callback` or pass the keyword argument `adaptive=false` (removing the need for setting `abstol` and `reltol`) to `solve` to disable error-based step size control and use CFL-based step size control instead.

Reply:
It doesn't make a difference right now, so I'll go ahead and remove it.

Reply:
Removing it is good; the CFL callback will indeed do nothing for adaptive methods right now, see https://github.com/trixi-framework/Trixi.jl/blob/63ba0f9799a7cf6712af3f8e9731d54eeb7a3c3e/src/callbacks_step/stepsize.jl#L67